designer/deployment-manager-api/src/main/scala/pl/touk/nussknacker/engine/DeploymentManagerProvider.scala

```diff
@@ -25,7 +26,7 @@ trait DeploymentManagerProvider extends NamedServiceProvider {
 
   def metaDataInitializer(config: Config): MetaDataInitializer
 
-  def scenarioPropertiesConfig(config: Config): Map[String, ScenarioPropertyConfig] = Map.empty
+  def scenarioPropertiesConfig(config: Config): ScenarioProperties = ScenarioProperties.empty()
```

Comment:
Let's not change this.
If we change it, we need to mention that in the migration guide, but I can't see benefits from providing this property URL programmatically in the Deployment Manager. It is overkill.

Reply:
If you're referring to changing the output type, I  absolutely don't think that leaving this as Map[...] is the good thing to do. It doesn't only make it weird from a domain standpoint where we have a domain object, which would be a natural way to use here. We will also mix up different abstraction levels in other parts of code where sometimes this Map, which is in reality a part of `ScenarioProperties`, will be mixed with `ScenarioProperties` themselves. It will also meddle with future changes to scenario properties. I think having this as a more separate, typed domain entity is way better.

Reply:
I'm not saying that you have to return a map. You can wrap it with some VO if you want, but please don't mix URL pointing to documentation in one place of application with list of properties that user have to fill to run it.

Also, follow the YAGNI and KISS rules. You currently don't have the requirement to make this URL dynamically changed, it is static, configured in the config, so don't expose this ability in the Deployment Manager API.